.github/workflows/build.yaml

```diff
@@ -0,0 +1,34 @@
+name: Build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - releases/**
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build Check
+    runs-on: blacksmith-4vcpu-ubuntu-2204
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: 1.81.0
+
+      - name: Rust Cache
+        uses: useblacksmith/rust-cache@v3
+        with:
+          shared-key: "cache"
+
+      - name: Run build
+        run: RUSTFLAGS="-D warnings" cargo build --locked --release
```

Comment:
we already have a compile gh action, should modify that instead of creating a duplicate one